src/Controller/MainController.java

```diff
@@ -74,6 +74,7 @@ public static void initialise()
                 CipherInputStream cipherInputStream = new CipherInputStream(new BufferedInputStream(new FileInputStream(fileName)), cipher);
                 ObjectInputStream inputStream = new ObjectInputStream(cipherInputStream);
                 SealedObject sealedObject = (SealedObject) inputStream.readObject();
+                inputStream.close();
```

Comment:
As I mentioned in class, the appropriate way to handle this is by managing the resource in a try-with-resources.

Reply:
But it is already using try-catch.

Reply:
That may be the case, but try-with-resources is the better construct for this.  There are several things inside of the try block that do not really need to be there and if you rework it as a try-with-resources it will be apparent.